src/Stack/Options.hs

```diff
@@ -682,7 +683,12 @@ initOptsParser :: Parser InitOpts
 initOptsParser =
     InitOpts <$> solver <*> omitPackages
              <*> overwrite <*> fmap not ignoreSubDirs
+             <*> searchDirs
   where
+    searchDirs =
+      many (textArgument
+              (metavar "SEARCH-DIRECTORIES" <>
+               help "Directories which are searched for cabal files. If non specified, uses current directory."))
```

Comment:
What you have is fine except for a spelling correction (non -> none). Since init does not have too many options, the help text is not too long therefore we do not need to be worried about making the help too concise.